pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,661 @@
+		retryAfterSeconds = 1
+		retryLater = true
+		createError = nil
+		return
+	}
+
+	if retryAfterCreateSeconds, shouldRetry := kerrors.SuggestsClientDelay(createError); shouldRetry {
+		retryAfterSeconds = retryAfterCreateSeconds
+		retryLater = true
+		createError = nil
+		return
+	}
+
+	if kerrors.IsForbidden(createError) {
+		if parent == h.homePrefix {
+			// If the parent itself is the home root, which is created by default => return the error.
+			return
+		}
+
+		grandParent, parentName := parent.Split()
+		// Check in the grand parent if the parent itself exists
+		if parent, err2 := h.getClusterWorkspace(ctx, grandParent, parentName); err2 == nil {
+			if parent.Status.Phase == tenancyv1alpha1.ClusterWorkspacePhaseReady {
+				// if we received 403 but the parent exists and is ready,
+				// there's no reason to wait more => return the error.
+				return
+			}
+
+			if workspaceUnschedulable(parent) {
+				// if we received 403 since the parent exists but cannot be scheduled,
+				// there's no reason to wait more => return the error.
+				return
+			}
+
+			// We have to wait for the parent workspace to be Ready before trying to create the child workspace
+			retryAfterSeconds = h.creationDelay
+			retryLater = true
+			createError = nil
+			return
+		}
+
+		// The parent simply probably does not exist => create it first
+		_, workspaceType := h.needsAutomaticCreation(parent)
+		retryLater, retryAfterSeconds, createError = h.tryToCreate(ctx, userName, parent, workspaceType)
+		return
+	}
+
+	// Return the error
+	return
+}
+
+// searchHomeWorkspaceRBACLocally searches for the expected RBAC resources associated to a Home workspace
+// in the local informers.
+func (h *homeWorkspaceHandler) searchHomeWorkspaceRBACLocally(homeWorkspace logicalcluster.Name, userName string) (found bool, err error) {
+	if h.mocks.searchHomeWorkspaceRBACLocally != nil {
+		return h.mocks.searchHomeWorkspaceRBACLocally(homeWorkspace, userName)
+	}
+
+	parentWorkspace, homeWorkspaceName := homeWorkspace.Split()
+
+	for _, search := range []searchFunc{h.searchLocalClusterRole, h.searchLocalClusterRoleBinding} {
+		for _, check := range []struct {
+			workspace logicalcluster.Name
+			name      string
+		}{
+			{
+				workspace: homeWorkspace,
+				name:      homeOwnerRBACName,
+			},
+			{
+				workspace: parentWorkspace,
+				name:      homeOwnerRBACPrefix + homeWorkspaceName,
+			},
+		} {
+			if ok, err := search(check.workspace, check.name); err != nil || !ok {
+				return ok, err
+			}
+		}
+	}
+
+	return true, nil
+}
+
+// createHomeRBAC uses an external client to create the RBAC resources related to a Home workspace
+// Note: it only create a cluster role or cluster role binding if it doesn't exist.
+// For now it doesn't try to detect and update (or recreate) RBAC resources that might be obsolete.
+func (h *homeWorkspaceHandler) createHomeRBAC(ctx context.Context, userName string, homeWorkspace logicalcluster.Name) error {
```

Comment:
did we ever think about custom authz for home workspaces? All this complexity is super ugly.

Reply:
Done in commit 0cb62c3aa30411ef67a7350b9ab85dec02c3e210